scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/goals/CardinalityGoal.java

```diff
@@ -107,32 +107,8 @@ protected CardinalityGoal fill(CardinalityGoal goal) {
       if (occurrenceRange != null) {
         goal.occurrenceRange = occurrenceRange;
       }
-      if(isUnsatisfiableDurationType()){
-        throw new IllegalArgumentException("Goal is incorrectly parametrized: activity creation template has zero-duration while duration objective is non-zero");
-      }
-      if(isUnsatisfiable()){
-        throw new IllegalArgumentException("Goal is incorrectly parametrized: minimum duration x minimum occurence > maximum duration");
-      }
       return goal;
     }
-
-    public boolean isUnsatisfiableDurationType(){
-      return (thereExists.getDurationRange() != null &&
-              thereExists.getDurationRange().isSingleton() &&
-              thereExists.getDurationRange().start.isZero() &&
-              durationRange.start.longerThan(Duration.ZERO)
-      );
-    }
-
-    public boolean isUnsatisfiable(){
-      return this.durationRange != null &&
-             occurrenceRange != null &&
-             this.thereExists.getDurationRange() != null &&
-             this.thereExists.getDurationRange().start
-                 .times(occurrenceRange.getMinimum())
-                 .longerThan(durationRange.end);
-    }
```

Comment:
Why a check of this flavor is not possible anymore ?

Reply:
They use `getDurationRange()`. I had to change the `Interval durationRange` field of `ActivityExpression` into `Expression<? extends Profile<?>> duration`. These unsatisfiability checks are sanity checks before the goal is run, but we can no longer check the duration before evaluation because the duration is lazy.